### PR TITLE
Add Llama 2

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

Add Llama 2.

### Corresponding documentation PR

https://github.com/explosion/spaCy/pull/12782

### Types of change

New feature.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran all tests in `tests` and `usage_examples/tests`, and all new and existing tests passed. This includes
  - all external tests (i. e. `pytest` ran with `--external`)
  - all tests requiring a GPU (i. e. `pytest` ran with `--gpu`)
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by rmitsch** [comment]
Ugh, we'd need to set up an access token secret for HuggingFace on BuildKite so we can download LLama 2...

File: spacy_llm/models/hf/llama2.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        return (
+            {
+                **default_cfg_init,
+                "trust_remote_code": True,
+            },
+            default_cfg_run,
+        )
+
+
+@registry.llm_models("spacy.Llama2.v1")
```
Comment (svlandeg) [inline spacy_llm/models/hf/llama2.py:60]: I think it's our best option, but `Llama2.v1` is making my head hurt a little...

File: spacy_llm/models/hf/llama2.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+def llama2_hf(
+    name: Llama2.MODEL_NAMES,
+    config_init: Optional[Dict[str, Any]] = SimpleFrozenDict(),
+    config_run: Optional[Dict[str, Any]] = SimpleFrozenDict(),
+) -> Callable[[Iterable[str]], Iterable[str]]:
+    """Generates Llama 2 instance that can execute a set of prompts and return the raw responses.
+    name (Literal): Name of the Llama 2 model. Has to be one of Llama2.get_model_names().
+    config_init (Optional[Dict[str, Any]]): HF config for initializing the model.
+    config_run (Optional[Dict[str, Any]]): HF config for running the model.
+    RETURNS (Callable[[Iterable[str]], Iterable[str]]): Falcon instance that can execute a set of prompts and return
```
Comment (svlandeg) [inline spacy_llm/models/hf/llama2.py:70]:
```suggestion
    RETURNS (Callable[[Iterable[str]], Iterable[str]]): Llama2 instance that can execute a set of prompts and return
```

**Comment by rmitsch** [comment]
> We didn't add this yet to https://github.com/explosion/spaCy/pull/12782, right? We still need to make a new PR?

[We did](https://github.com/explosion/spaCy/pull/12782/files#diff-8a76a363f57b6555db91f2da7a2f9b80bce2913a4c4133d108b6db6b01cbd64fR1189).

File: spacy_llm/models/hf/llama2.py (hunk 1 above)
Comment (rmitsch) [reply spacy_llm/models/hf/llama2.py:60]: I think it's not that bad, actually. I'm more annoyed by the changing capitalization, but :man_shrugging:

**Pull request merged**